plugins/modules/firewalld_info.py

```diff
@@ -0,0 +1,392 @@
+                            type: bool
+                            sample: false
+                        forward_ports:
+                            description:
+                              - A list of forwarding port pair with protocol.
+                            returned: success
+                            type: list
+                            sample:
+                              - "icmp"
+                              - "ipv6-icmp"
+                        source_ports:
+                            description:
+                              - A list of network source port with protocol.
+                            returned: success
+                            type: list
+                            sample:
+                              - - "30000"
+                                - "tcp"
+                              - - "30001"
+                                - "tcp"
+                        icmp_blocks:
+                            description:
+                              - A list of blocking icmp protocol.
+                            returned: success
+                            type: list
+                            sample:
+                              - "echo-request"
+                        rich_rules:
+                            description:
+                              - A list of rich language rule.
+                            returned: success
+                            type: list
+                            sample:
+                              - "rule protocol value=\"icmp\" reject"
+                              - "rule priority=\"32767\" reject"
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from distutils.version import StrictVersion
+
+
+try:
+    import dbus
+    HAS_DBUS = True
+except ImportError:
+    HAS_DBUS = False
+
+try:
+    import firewall.client as fw_client
+    import firewall.config as fw_config
+    HAS_FIREWALLD = True
+except ImportError:
+    HAS_FIREWALLD = False
+
+
+def get_version():
+    return fw_config.VERSION
+
+
+def get_active_zones(client):
+    return client.getActiveZones().keys()
+
+
+def get_all_zones(client):
+    return client.getZones()
+
+
+def get_default_zone(client):
+    return client.getDefaultZone()
+
+
+def get_zone_settings(client, zone):
+    return client.getZoneSettings(zone)
+
+
+def get_zone_target(zone_settings):
+    return zone_settings.getTarget()
+
+
+def get_zone_icmp_block_inversion(zone_settings):
+    return zone_settings.getIcmpBlockInversion()
+
+
+def get_zone_interfaces(zone_settings):
+    return zone_settings.getInterfaces()
+
+
+def get_zone_sources(zone_settings):
+    return zone_settings.getSources()
+
+
+def get_zone_services(zone_settings):
+    return zone_settings.getServices()
+
+
+def get_zone_ports(zone_settings):
+    return zone_settings.getPorts()
+
+
+def get_zone_protocols(zone_settings):
+    return zone_settings.getProtocols()
+
+
+# This function suports python-firewall 0.9.0(or later).
```

Comment:
```suggestion
# This function supports python-firewall 0.9.0(or later).
```

Reply:
Add a note in documentation stating this limitation.

Reply:
Sorry for the typo... I have fixed it.